src/sql/workbench/contrib/searchViewPane/browser/searchWidget/searchWidgetService.ts

```diff
@@ -0,0 +1,297 @@
+	) {
+		super();
+		this.searchWidgets = new Map<string, any>();
+	}
+
+	getSearchWidget(VIEWLET_ID: string): any | undefined {
+		return this.searchWidgets.get(VIEWLET_ID);
+	}
+
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void {
+		if (!this.searchWidgets.get(params.VIEWLET_ID)) {
+			let searchWidget = this._register(this._instantiationService.createInstance(SimpleSearchWidget, container, <IViewExplorerSearchOptions>{
+				value: '',
+				replaceValue: undefined,
+				isRegex: false,
+				isCaseSensitive: false,
+				isWholeWords: false,
+				searchHistory: [],
+				replaceHistory: [],
+				preserveCase: false,
+				showSearchResultsPane: params.showSearchResultsPane,
+			}, pane, params.VIEWLET_ID));
+
+			this.searchWidgets.set(params.VIEWLET_ID, searchWidget);
+			if (params.showSearchResultsPane) {
+				this.registerSearchResultsView(params);
+
+				//register search results view specific stuff.
+				if (searchWidget.searchView) {
+					this._register(searchWidget.onDidHeightChange(() => searchWidget.searchView?.reLayout()));
+
+					this._register(searchWidget.onPreserveCaseChange(async (state) => {
+						if (searchWidget.searchView?.searchViewModel) {
+							searchWidget.searchView.searchViewModel.preserveCase = state;
+							await searchWidget.searchView.refreshTree();
+						}
+					}));
+
+					this._register(searchWidget.searchInput.onInput(() => searchWidget.searchView?.updateActions()));
+				}
+			}
+
+			if (params.onSearchSubmit) {
+				searchWidget.onSearchSubmit(options => params.onSearchSubmit(options));
+
+				searchWidget.searchInput.onDidOptionChange((options) => params.onSearchSubmit(options));
+			}
+
+			if (params.onSearchCancel) {
+				searchWidget.onSearchCancel((focus) => params.onSearchCancel(focus));
+			}
+		}
+	}
+
+	registerSearchResultsView(params: ISearchViewPaneOptions) {
+		let viewDescriptors = [];
+		switch (params.VIEWLET_ID) {
+			case 'workbench.view.notebooks': viewDescriptors.push({
+				id: NotebookSearchResultsView.ID,
+				name: localize('notebookExplorer.searchResults', "Search Results"),
+				ctorDescriptor: new SyncDescriptor(NotebookSearchResultsView),
+				weight: 100,
+				canToggleVisibility: true,
+				hideByDefault: false,
+				order: 0,
+				collapsed: true
+
+			});
+				break;
+			default: break;
+		}
+
+		const container = this.viewDescriptorService.getViewContainerById(params.VIEWLET_ID);
+		Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews(viewDescriptors, container);
+	}
+
+	getSearchResultsView(parentConatiner: ViewPaneContainer): ISearchResultsView {
+		switch (parentConatiner.getId()) {
+			case 'workbench.view.notebooks': return <NotebookSearchResultsView>parentConatiner.getView(NotebookSearchResultsView.ID);
+			default: return undefined;
+		}
+	}
+}
+
+const configurationRegistry = Registry.as<IConfigurationRegistry>(ConfigExtensions.Configuration);
```

Comment:
Do you have any plan for making sure that this is kept in sync with the changes made to the vs code search implementation? How are we going to make sure that if they change any of these settings on their side we're at least aware so we can make a decision whether to update ours too?

Reply:
I didn't want to keep it in sync with vs code search but since we're overriding from vscode search, I get your point. I'll double check and make sure that even if they make any changes that it wouldn't break us. As for getting the updates, unless there is an ask from users for something that's supported there and missing in this, we shouldn't be updating from them. @chlafreniere  Can you provide more perspective here.